gmeta/src/lib.rs

```diff
@@ -85,15 +93,42 @@ impl Types for () {
 
 impl MetadataRepr {
     pub fn bytes(&self) -> Vec<u8> {
-        self.encode()
+        // Append language ID and version as a preamble
```

Comment:
Create vec with capacity, extend from u16 instead of manual gets + use `.encode_to(bytes)`

Reply:
Or you can directly encode `(lang, version, registry).encode()`. Tuple just concats bytes